core/src/main/java/com/linecorp/armeria/client/retry/RetryingClient.java

```diff
@@ -75,20 +74,20 @@ private static void setDeadlineOfThisRequest(ClientRequestContext ctx) {
         }
     }
 
-    @VisibleForTesting
-    Backoff newBackoff() {
-        Backoff backoff = backoffSupplier.get();
-        if (!backoff.as(AttemptLimitingBackoff.class).isPresent()) {
-            backoff = backoff.withMaxAttempts(defaultMaxAttempts);
+    private void initRetryStrategyState(ClientRequestContext ctx) {
+        final RetryStrategyState retryStrategyState = ctx.attr(LAST_RETRY_STRATEGY_STATE).get();
+        if (retryStrategyState != null) {
+            retryStrategyState.init();
+        } else {
+            ctx.attr(LAST_RETRY_STRATEGY_STATE).set(new RetryStrategyState(defaultMaxAttempts));
```

Comment:
`ctx.attr(...)` creates a new `Attribute` instance when no attribute is set. We could extract it into a variable:

```java
final Attribute<RetryStrategyState> attr = ctx.attr(...);
final RetryStrategyState retryStrategyState = attr.get();
if (...) {
    ...
} else {
    attr.set(new RetryStrategyState(...));
}
```